pkg/test/framework/components/echo/kube/instance.go

```diff
@@ -313,6 +319,35 @@ func workloadHasSidecar(cfg echo.Config, podName string) bool {
 	return true
 }
 
+// Update the workload endpoint to the config map created by `Istiod`.
+// External workloads have no endpoint registered in k8s, so the traffic sent to these non-k8s workloads
+// won't be picked up by Prometheus. We update the workload entries' endpoints to a config map so that
+// Prometheus could mount the map and read the endpoints.
+func updateWorkloadEndpoint(client kubernetes.Interface, name, endpoint, service, version string) error {
+	currentCM, err := client.CoreV1().ConfigMaps("istio-system").Get(context.TODO(),
+		"file-sd-config", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if currentCM.Data == nil {
+		currentCM.Data = make(map[string]string)
+	}
+
+	staticConfig := `
+- targets:
+  - %s
+  labels:
```

Comment:
i wouldn't worry about doing any labeling. Rely on that information to come from the proxies themselves. We don't need to match the k8s discovery stuff (in fact, we've tried to remove the extra stuff that adds in previous iterations, etc.)

Reply:
Understood!